lib/saml.js

```diff
@@ -182,31 +182,54 @@ SAML.prototype.certToPEM = function (cert) {
 SAML.prototype.checkSAMLStatus = function (xmlDomDoc) {
 	var status = {StatusCodeValue:null, StatusMessage:null, StatusDetail:null}
 
-	var statusCodeValueNode = xmlCrypto.xpath(xmlDomDoc, "//*[local-name(.)='StatusCode']")[0];
-	if(statusCodeValueNode)
-	{
+	try{
+    var statusCodeValueNode = xmlCrypto.xpath(xmlDomDoc, "//*[local-name(.)='StatusCode']")[0];
+  }catch(err){
+    // Error Handling
+  }
+
+	if(statusCodeValueNode != null || undefined){
```

Comment:
use `typeof` to check for `undefined`

Also, you have to compare it against `statusCodeValueNode`.

eg:
```
if (a = true ||  1) // bad
if (a = true || a = 1) // good
```